spring-cloud-starter-stream-processor-grpc/src/main/java/org/springframework/cloud/stream/app/grpc/processor/GrpcProcessorConfiguration.java

```diff
@@ -0,0 +1,128 @@
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.grpc.processor;
+
+import com.google.protobuf.Empty;
+import io.grpc.Channel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.stub.StreamObserver;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.annotation.StreamListener;
+import org.springframework.cloud.stream.app.grpc.support.MessageUtils;
+import org.springframework.cloud.stream.app.grpc.support.ProtobufMessageBuilder;
+import org.springframework.cloud.stream.messaging.Processor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessagingException;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @author David Turanski
+ **/
+
+@EnableBinding(Processor.class)
+@EnableConfigurationProperties(GrpcProperties.class)
+public class GrpcProcessorConfiguration {
+
+	@Autowired
+	private Channel grpcChannel;
+
+	@Autowired
+	private GrpcProperties properties;
+
+	@Autowired
+	private Processor channels;
+
+	@Autowired
+	private ProcessorGrpc.ProcessorStub processorStub;
+
+	@Bean
+	public ProcessorGrpc.ProcessorStub processorStub() {
+		return ProcessorGrpc.newStub(grpcChannel);
+	}
+
+	@Bean
+	@ConditionalOnProperty(name = "grpc.host")
+	public Channel grpcChannel(GrpcProperties properties) {
+		ManagedChannelBuilder<?> managedChannelBuilder = ManagedChannelBuilder
+			.forAddress(properties.getHost(), properties.getPort()).usePlaintext(properties.isPlainText())
+			.directExecutor();
+		if (properties.getIdleTimeout() > 0) {
+			managedChannelBuilder = managedChannelBuilder.idleTimeout(properties.getIdleTimeout(), TimeUnit.SECONDS);
+		}
+		if (properties.getMaxMessageSize() > 0) {
+			managedChannelBuilder = managedChannelBuilder.maxInboundMessageSize(properties.getMaxMessageSize());
+		}
+		return managedChannelBuilder.build();
+	}
+
+	@StreamListener(Processor.INPUT)
+	public void process(final Message<?> request) {
+		ProtobufMessageBuilder protobufMessageBuilder = new ProtobufMessageBuilder();
+
+		org.springframework.cloud.stream.app.grpc.message.Message protobufMessage = properties.isIncludeHeaders() ?
+			protobufMessageBuilder.fromMessage(request).build() :
+			protobufMessageBuilder.withPayload(request.getPayload()).build();
+
+		processorStub
```

Comment:
The only comment I have for this PR is regarding this line @dturanski. My understanding is that since this is an asynchronous call, the remote endpoint could potentially return messages on different order. Say we receive M1,M2 and call it async but the server replies M2,M1. 

I don't see this as an issue to be honest, but it may be worth either letting users know, or allow this to be configurable via blockingStub or Stub depending on the running mode. Thoughts?

Reply:
Yeah. Strict order is important in some use cases. I'll take a look at switching to BlockingStub or making it configurable.